src/NodeApi/DotNetHost/NativeHost.cs

```diff
@@ -48,15 +50,18 @@ public static napi_value InitializeModule(napi_env env, napi_value exports)
     {
         Trace($"> NativeHost.InitializeModule({env.Handle:X8}, {exports.Handle:X8})");
 
-        JSRuntime runtime = new NodejsRuntime();
-        using JSValueScope scope = new(JSValueScopeType.NoContext, env, runtime);
```

Comment:
>JSValueScope [](http://example.com/codeflow?start=14&length=12)

JSValueScoper must be bound to the call stack. Storing it as a field defeats the purpose as it now will say that `napi_value` created in this scope are valid, while in practice they will point to deleted memory.

Reply:
I understand and I'll consider a different approach.

This is tricky because the `JSRuntime` instance must be passed to callbacks somehow, and that cannot be through instance data because the `JSRuntime` instance is needed to call the API to get instance data. The alternative is probably to use another thread-local static for the `JSRuntime` instance.

Reply:
Having a thread local for `JSRuntime` does not sound that bad.

Reply:
As discussed, we'll defer this issue for now, and address it in another change soon. If you are adding a `JSEnvironment` class or something like that, it might make sense to use that to resolve this issue. Or if that doesn't work I'll consider the thread-local `JSRuntime` mentioned above. But for now I'll merge this to unblock your work.